### ISSUE TITLE
The animation occasionally fails to change from startValue to value.

### ISSUE THREAD

**Issue by ivonzhang** [almond-bongbong/react-slot-counter#64]
Hi，there are two problems I found。For example, when changing from 29 to 1, the animation stops at 9 instead of 1. Additionally, when trying to decrement 29 to 1 sequentially, sequentialAnimationMode is ineffective.

**Comment by ivonzhang** [comment]
My code:
```typescript
<SlotCounter
        ref={counterRef}
        numberSlotClassName={numberSlotClassName}
        startValue={startValue}
        value={value}
        // duration={1.6}
        autoAnimationStart={false}
        sequentialAnimationMode
        direction="bottom-up"
        useMonospaceWidth
        startValueOnce
        // animateOnVisible={{ triggerOnce: true, rootMargin: '0px 0px -100px 0px' }}
      />
``` 
<details><summary>Details</summary>
<p>

then, in react ，useEffect(() => { if (isInView) counterRef.current?.startAnimation(); })。

</p>
</details>

**Comment by almond-bongbong** [comment]
Thank you for reporting these issues and providing the detailed example. We truly appreciate your feedback and will investigate these problems right away.

**Comment by almond-bongbong** [comment]
@ivonzhang After reviewing your report, it seems the second issue may occur when using the direction="bottom-up" prop. We’ll work on improving this behavior.

Additionally, could you please provide a reproducible demo for the first issue, as well as the version of react-slot-counter you’re currently using?

**Comment by ivonzhang** [comment]
> @ivonzhang After reviewing your report, it seems the second issue may occur when using the direction="bottom-up" prop. We’ll work on improving this behavior.
> 
> Additionally, could you please provide a reproducible demo for the first issue, as well as the version of react-slot-counter you’re currently using?

the version is 3.0.2.

**Comment by ivonzhang** [comment]
> @ivonzhang After reviewing your report, it seems the second issue may occur when using the direction="bottom-up" prop. We’ll work on improving this behavior.
> 
> Additionally, could you please provide a reproducible demo for the first issue, as well as the version of react-slot-counter you’re currently using?

The first problem will occur on some real mobile devices occasionally... maybe you should use some low-end devices to try...

**Comment by almond-bongbong** [comment]
The issue has been resolved in version 3.0.3. If the problem persists, please reopen or create a new issue. Thank you once again.
